src/Index/Index.php

```diff
@@ -216,10 +287,53 @@ public function unserialize($serialized)
     public function serialize()
     {
         return serialize([
-            'definitions' => $this->definitions,
+            'namespaceDefinitions' => $this->namespaceDefinitions,
             'references' => $this->references,
             'complete' => $this->complete,
             'staticComplete' => $this->staticComplete
         ]);
     }
+
+    /**
+     * Registers a definition to the global definitions index if it is global
+     *
+     * @param string $fqn The fully qualified name of the symbol
+     * @param Definition $definition The Definition object
+     * @return void
+     */
+    private function setGlobalDefinition(string $fqn, Definition $definition)
+    {
+        if ($definition->isGlobal) {
+            $this->globalDefinitions[$fqn] = $definition;
+        }
+    }
+
+    /**
+     * @param string $fqn
+     * @return string The namespace extracted from the given FQN
+     */
+    private function extractNamespace(string $fqn): string
+    {
+        foreach (['::', '->'] as $operator) {
+            if (false !== ($pos = strpos($fqn, $operator))) {
+                return substr($fqn, 0, $pos);
+            }
+        }
+
+        return $fqn;
+    }
+
+    /**
+     * Returns the Definitions that are in the given namespace
+     *
+     * @param string $namespace
+     * @return Definition[]
+     */
+    private function doGetDefinitionsForNamespace(string $namespace): array
```

Comment:
To avoid logic duplication because I'm using it twice in this class : 
- https://github.com/nicolasmure/php-language-server/blob/6ba3d6c0c52cd9177d68a3701ef81159dda13aa9/src/Index/Index.php#L130
- https://github.com/nicolasmure/php-language-server/blob/6ba3d6c0c52cd9177d68a3701ef81159dda13aa9/src/Index/Index.php#L145

Moreover, this private function returns an array, in opposition to the public function `getDefinitionsForNamespace` which returns a generator. This array is then used in other public functions of this class to yield definitions.